pipeline_utils/pipeline_deploy.py

```diff
@@ -0,0 +1,406 @@
+            return YAMLClass(data_yaml).to_json(**kwargs)
+
+        return
+
+
+    def _post_patch_file(self, type):
+        """
+            'Software', 'FileFormat', 'FileReference'
+        """
+        logger.info(f'@ {type}...')
+
+        # Set general variables
+        filepath_ = f'{self.repo}/{self.filepath[type]}'
+
+        # Check
+        if not os.path.isfile(filepath_):
+            logger.error(f'WARNING: {self.filepath[type]} not found in {self.repo}, skipping...')
+            return
+
+        # Read YAML file and create JSON objects from documents in file
+        for d in yaml_parser.load_yaml(filepath_):
+            # creating JSON object
+            d_ = self._yaml_to_json(
+                        d, self.object_[type],
+                        institution=self.institution,
+                        project=self.project
+                        )
+            # post/patch object
+            if d_: self._post_patch_json(d_, type)
+
+
+    def _post_patch_folder(self, type):
+        """
+            'Workflow', 'MetaWorkflow'
+        """
+        logger.info(f'@ {type}...')
+
+        # Set general variables
+        filepath_ = f'{self.repo}/{self.filepath[type]}'
+
+        # Check
+        if not os.path.isdir(filepath_):
+            logger.error(f'WARNING: {self.filepath[type]} not found in {self.repo}, skipping...')
+            return
+
+        # Create JSON objects
+        for fn in glob.glob(f'{filepath_}/*.yaml'):
+            for d in yaml_parser.load_yaml(fn):
+                # creating _yaml_to_json **kwargs
+                kwargs_ = {
+                    'version': self.version,
+                    'institution': self.institution,
+                    'project': self.project
+                }
+                if type == 'Workflow':
+                    kwargs_.setdefault(
+                        'wflbucket_url', f's3://{self.wfl_bucket}/{self.pipeline}/{self.version}'
+                    )
+                # creating JSON object
+                d_ = self._yaml_to_json(
+                            d, self.object_[type],
+                            **kwargs_
+                            )
+                # post/patch object
+                if d_: self._post_patch_json(d_, type)
+
+
+    def _post_patch_wfl(self, type='WFL'):
+        """
+        """
+        logger.info('@ Workflow Description...')
+
+        # Set general variables
+        filepath_ = f'{self.repo}/{self.filepath[type]}'
+        upload_ = f'{filepath_}/upload'
+        account_ = f'{self.account}.dkr.ecr.{self.region}.amazonaws.com'
+        update_ = {
+            'ServerSideEncryption': 'aws:kms',
+            'SSEKMSKeyId': self.kms_key_id
+            }
+
+        # Check
+        if not os.path.isdir(filepath_):
+            logger.error(f'WARNING: {self.filepath[type]} not found in {self.repo}, skipping...')
+            return
+
+        # Create s3 object
+        s3 = boto3.resource('s3')
```

Comment:
So here you're assuming some external credential setup beyond those needed for the portal you may want to validate. I would just consider writing a function that at startup of this sanity checks that everything is set, meaning:

1. `AWS_ACCESS_KEY_ID`
2. `AWS_SECRET_ACCESS_KEY`
3. `AWS_SESSION_TOKEN` (if the key ID begins with "ASIA", as that prefix indicates short term credentials)
4. `GLOBAL_ENV_BUCKET`
5. `S3_ENCRYPT_KEY`